### PR TITLE
Change vaultApy and lpApy to use the same calculation method

### PR DESCRIPTION
**Pull request by MazyGio**
## Problem
LP APY and Variable APY differ significantly from each other, even in markets with no open long/short positions.

## Explanation
LP APY looks at `lpSharePrice` on a checkpoint-by-checkpoint basis, within a 7-day timeframe.
Yield Source APY looks at `vaultSharePrice` on a block-by-block basis, within a 2-day timeframe.

This difference in methods and time frames results in different rates.

## Solution
This PR changes both to update on a block-by-block basis, within a 1-day timeframe

Hyperdrive_-_DeFi_Yield__your_way

Hyperdrive_-_DeFi_Yield__your_way

### CODE REVIEW

File: packages/hyperdrive-js-core/src/hyperdrive/base/ReadHyperdrive.ts [hunk 1]
```diff
@@ -230,24 +228,15 @@ export class ReadHyperdrive extends ReadModel {
     const { vaultSharePrice: currentVaultSharePrice } =
       await this.getPoolInfo(options);
 
-    const timeRange = currentBlock.timestamp - startBlock.timestamp; // bigint
+    const timeFrame = BigInt(currentBlock.timestamp - startBlock.timestamp);
```
Comment (ryangoree) [inline packages/hyperdrive-js-core/src/hyperdrive/base/ReadHyperdrive.ts:231]: These should already be bigints, so we can remove the constructor:
```suggestion
    const timeFrame = currentBlock.timestamp - startBlock.timestamp;
```

**Review by DannyDelott — APPROVED** [review]

**Review by ryangoree — APPROVED** [review]

File: packages/hyperdrive-js-core/src/hyperdrive/base/ReadHyperdrive.ts (hunk 1 above)
Comment (MazyGio) [reply packages/hyperdrive-js-core/src/hyperdrive/base/ReadHyperdrive.ts]: Ah, good catch! my IDE was playing games on me. Fixed it here and on the lpApy function.

File: packages/hyperdrive-js-core/src/hyperdrive/base/ReadHyperdrive.ts [hunk 2]
```diff
@@ -207,20 +206,19 @@ export class ReadHyperdrive extends ReadModel {
...
+    // If the 24 hour rate doesn't exist, assume the pool was initialized less
+    // than 24 hours ago and try to get the all-time rate
     const { blockNumber: initializationBlock } =
       await this.getInitializationBlock();
-    if (initializationBlock && startBlockNumber < initializationBlock) {
-      throw new HyperdriveSdkError(
-        `Unable to calculate yield source APY. Attempted to fetch data from block ${startBlockNumber}, but the pool was initialized at block ${initializationBlock}.`,
-      );
+    if (initializationBlock && initializationBlock > startBlockNumber) {
+      startBlockNumber = initializationBlock;
```
[the remark this answers is not in the document]
Comment (MazyGio) [reply packages/hyperdrive-js-core/src/hyperdrive/base/ReadHyperdrive.ts:215]: Updated!

File: packages/hyperdrive-js-core/src/hyperdrive/base/ReadHyperdrive.ts [hunk 3]
```diff
@@ -2025,19 +2009,16 @@ export class ReadHyperdrive extends ReadModel {
...
  * r = (p_1 / p_0) ^ (1 / t) - 1
  */
-function calculateLpApy({
-  startingLpSharePrice,
-  endingLpSharePrice,
-  timeFrame,
+function calculateApyFromSharePrice({
+  startingSharePrice,
+  endingSharePrice,
+  timeFrame, // seconds
```
[the remark this answers is not in the document]
Comment (MazyGio) [reply packages/hyperdrive-js-core/src/hyperdrive/base/ReadHyperdrive.ts]: Sounds good, updated!

**Pull request merged**